src/app/tree-grid/tree-grid-row-reorder/tree-grid-row-reorder.component.ts

```diff
@@ -0,0 +1,80 @@
+import { Component, ViewChild } from "@angular/core";
+import { IgxDropEventArgs, IgxTreeGridComponent, IgxTreeGridRowComponent, Point } from "igniteui-angular";
+import { EMPLOYEE_FLAT_DATA } from "../data/employees-flat";
+
+@Component({
+    selector: "tree-grid-row-reorder",
+    styleUrls: ["tree-grid-row-reorder.component.scss"],
+    templateUrl: "tree-grid-row-reorder.component.html"
+})
+export class TreeGridRowReorderComponent {
+    @ViewChild(IgxTreeGridComponent, { read: IgxTreeGridComponent })
+    public treeGrid: IgxTreeGridComponent;
+
+    public localData = [];
+    constructor() {
+        this.localData = EMPLOYEE_FLAT_DATA();
+    }
+
+    public rowDragStart(args: any): void {
+        const targetRow: IgxTreeGridRowComponent = args.dragData;
+        if (targetRow.expanded) {
+            this.treeGrid.collapseRow(targetRow.rowID);
+        }
+    }
+
+    public dropInGrid(args: IgxDropEventArgs): void {
+        args.cancel = true;
+        const draggedRow: IgxTreeGridRowComponent = args.dragData;
+        const event = args.originalEvent;
+        const cursorPosition: Point = { x: event.pageX, y: event.pageY };
+        this.moveRow(draggedRow, cursorPosition);
+    }
+
+    private moveRow(draggedRow: IgxTreeGridRowComponent, cursorPosition: Point): void {
+        const row: IgxTreeGridRowComponent = this.catchCursorPosOnElem(this.treeGrid.rowList.toArray(), cursorPosition);
+        if (!row) { return; }
+        if (row.rowData.ParentID === -1) {
+            this.performDrop(draggedRow, row).ParentID = -1;
+        } else {
+            if (row.rowData.ParentID === draggedRow.rowData.ParentID) {
+                this.performDrop(draggedRow, row);
+            } else {
+                const rowIndex = this.getRowIndex(row);
```

Comment:
This call should be:
```suggestion
                const rowIndex = this.getRowIndex(draggedRow.rowData);
```
You want to set the **draggedRow**'s parentID to be the same as one where it's dropped (so they become siblings)